### PR TITLE
SQL: Grammar tweak for number declarations

### LINKED ISSUE #33765: SQL: improvements to negative numbers
Currently negative numbers are handled as `Not(number)` which complicates the plans considerably and introduces bugs in folding (such as ordering).

### PR DESCRIPTION
**Pull request by costin**
Consider plus and minus as part of a number declaration (to avoid the
minus be treated as a negation).

Close #33765

### CODE REVIEW

**Review by astefan — APPROVED** [review]
LGTM

File: x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/parser/ExpressionTests.java [hunk 1]
```diff
@@ -9,7 +9,6 @@
 import org.elasticsearch.xpack.sql.expression.Expression;
 import org.elasticsearch.xpack.sql.expression.Literal;
 import org.elasticsearch.xpack.sql.expression.function.UnresolvedFunction;
-import org.elasticsearch.xpack.sql.expression.function.scalar.arithmetic.Neg;
 import org.elasticsearch.xpack.sql.type.DataType;
 
 public class ExpressionTests extends ESTestCase {
```
Comment (matriv) [inline x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/parser/ExpressionTests.java]: Would be nice to add also the same test for decimal expression, just for the "completeness".

**Review by matriv — APPROVED** [review]
LGTM with a comment.

File: x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/parser/ExpressionTests.java (hunk 1 above)
Comment (costin) [reply x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/parser/ExpressionTests.java]: Addressed that through https://github.com/elastic/elasticsearch/pull/33767/commits/0365332bdc9f8723201601d621b003f152b0acb4

**Pull request merged**